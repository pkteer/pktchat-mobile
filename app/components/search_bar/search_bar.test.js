// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {shallowWithIntl} from '@test/intl-test-helper';

import SearchBar from './index';

jest.useFakeTimers();

describe('SearchBar', () => {
    const baseProps = {
        onBlur: jest.fn,
        onFocus: jest.fn,
        onSearchButtonPress: jest.fn,
        onChangeText: jest.fn,
        onCancelButtonPress: jest.fn,
        onSelectionChange: jest.fn,
        testID: 'search_bar',
        backgroundColor: '#ffffff',
        placeholderTextColor: '#000000',
        titleCancelColor: '#aaaaaa',
        tintColorSearch: '#bbbbbb',
        tintColorDelete: '#cccccc',
        selectionColor: '#dddddd',
        inputStyle: {},
        containerStyle: {},
        cancelButtonStyle: {},
        autoFocus: true,
        placeholder: 'placeholder',
        cancelTitle: 'cancel title',
        returnKeyType: 'return-key-type',
        keyboardType: 'keyboard-type',
        autoCapitalize: 'auto-capitalize',
        inputHeight: 10,
        editable: true,
        blurOnSubmit: true,
        keyboardShouldPersist: true,
        value: 'value',
        keyboardAppearance: 'keyboard-appearance',
        showArrow: true,
        searchBarRightMargin: 5,
        searchIconSize: 1,
        backArrowSize: 1,
        deleteIconSize: 1,
        showCancel: true,
        containerHeight: 20,
    };

    test('should match snapshot', () => {
        const wrapper = shallowWithIntl(<SearchBar {...baseProps}/>);

        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
