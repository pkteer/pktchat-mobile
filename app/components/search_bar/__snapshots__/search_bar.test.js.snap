// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 20,
        "justifyContent": "flex-start",
        "overflow": "hidden",
      },
      Object {},
    ]
  }
  testID="search_bar"
>
  <ForwardRef(AnimatedComponentWrapper)
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "height": 10,
          "marginRight": 5,
        },
        Object {
          "marginLeft": 0,
        },
      ]
    }
  >
    <ForwardRef(Themed.SearchBar)
      allowFontScaling={true}
      autoCapitalize="auto-capitalize"
      autoCorrect={false}
      autoFocus={true}
      blurOnSubmit={true}
      cancelButtonProps={
        Object {
          "buttonTextStyle": Object {
            "color": "#aaaaaa",
            "fontSize": 14,
          },
        }
      }
      cancelButtonTitle="cancel title"
      cancelIcon={null}
      clearIcon={
        <CompassIcon
          name="close-circle"
          onPress={[Function]}
          size={18}
          style={
            Object {
              "color": "#cccccc",
            }
          }
          testID="search_bar.search.clear.button"
        />
      }
      containerStyle={
        Object {
          "backgroundColor": "#ffffff",
          "flex": 1,
          "paddingBottom": 0,
          "paddingTop": 0,
        }
      }
      disableFullscreenUI={true}
      editable={true}
      enablesReturnKeyAutomatically={true}
      inputContainerStyle={
        Object {
          "backgroundColor": undefined,
          "borderRadius": 2,
          "height": 10,
        }
      }
      inputStyle={
        Object {
          "backgroundColor": "transparent",
          "color": "#fff",
          "fontSize": 14,
          "height": 10,
          "marginLeft": 4,
          "marginTop": 0,
          "paddingTop": 0,
        }
      }
      keyboardAppearance="keyboard-appearance"
      keyboardType="keyboard-type"
      leftIconContainerStyle={
        Object {
          "marginLeft": 4,
          "width": 30,
        }
      }
      onBlur={[Function]}
      onCancel={[Function]}
      onChangeText={[Function]}
      onClear={[Function]}
      onFocus={[Function]}
      onSelectionChange={[Function]}
      onSubmitEditing={[Function]}
      placeholder="placeholder"
      placeholderTextColor="#000000"
      platform="ios"
      returnKeyType="return-key-type"
      searchIcon={
        <CompassIcon
          name="magnify"
          size={24}
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "color": "#bbbbbb",
                "top": 8,
              },
            ]
          }
        />
      }
      selectionColor="#dddddd"
      showCancel={true}
      testID="search_bar.search.input"
      underlineColorAndroid="transparent"
      value="value"
    />
  </ForwardRef(AnimatedComponentWrapper)>
</View>
`;
